optimum/neuron/modeling_base.py

```diff
@@ -12,600 +12,29 @@
-            cache_available = compile_cache.download_folder(model_cache_dir, model_cache_dir)
-        else:
-            cache_available = False
-
-        # load cache
-        if cache_available:
-            try:
-                neuron_model = cls.from_pretrained(model_cache_dir)
-                model = TasksManager.get_model_from_task(
-                    task=task,
-                    model_name_or_path=model_id,
-                    subfolder=subfolder,
-                    revision=revision,
-                    framework="pt",
-                    library_name=library_name,
-                    cache_dir=cache_dir,
-                    use_auth_token=use_auth_token,
-                    local_files_only=local_files_only,
-                    force_download=force_download,
-                    trust_remote_code=trust_remote_code,
-                )
-                # replace weights
-                neuron_model.replace_weights(weights=model)
-                return neuron_model
-            except Exception as e:
-                logger.warning(
-                    f"Found the cached artifacts but failed to re-load them with error: {e}. \n Falling back to recompilation."
-                )
-                cache_available = False
-
-        # compile
-        if not cache_available:
-            # compile
-            save_dir = TemporaryDirectory()
-            save_dir_path = Path(save_dir.name)
-            main_export(
-                model_name_or_path=model_id,
-                output=save_dir_path,
-                compiler_kwargs=compiler_kwargs,
-                task=task,
-                dynamic_batch_size=dynamic_batch_size,
-                cache_dir=cache_dir,
-                disable_neuron_cache=disable_neuron_cache,
-                compiler_workdir=compiler_workdir,
-                inline_weights_to_neff=inline_weights_to_neff,
-                optlevel=optlevel,
-                trust_remote_code=trust_remote_code,
-                subfolder=subfolder,
-                revision=revision,
-                force_download=force_download,
-                local_files_only=local_files_only,
-                use_auth_token=use_auth_token,
-                do_validation=False,
-                library_name=library_name,
-                **kwargs_shapes,
-            )
-            config = AutoConfig.from_pretrained(save_dir_path)
-
-        return cls._from_pretrained(save_dir_path, config, model_save_dir=save_dir)
-
-    def push_to_hub(
-        self,
-        save_directory: str,
-        repository_id: str,
-        private: Optional[bool] = None,
-        revision: Optional[str] = None,
-        use_auth_token: Union[bool, str] = True,
-        endpoint: Optional[str] = None,
-    ) -> str:
-        if isinstance(use_auth_token, str):
-            huggingface_token = use_auth_token
-        elif use_auth_token:
-            huggingface_token = HfFolder.get_token()
+        if hasattr(model, "device"):
```

Comment:
Is it necessary, for traced model setting all dummy device as CPU would be fine I think.

Reply:
As long as it exist, someone might want to call it, so I added a check.